### PR TITLE
Add npm and grunt, adjust spacings and some more styles

### PR DESCRIPTION
**Pull request by AmrAbdulrahman**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+node_modules
```
Comment (mabdurrahman) [inline .gitignore:1]: Why don't you add this to already existing root directory .gitignore ?

File: assets/css/styles.css [hunk 2]
```diff
@@ -32,6 +32,42 @@ a {
...
+  content: "";
+  position: absolute;
+  top: 0;
+  left: 0;
+  width: 100%;
+  height: 100%;
+  background-color: rgba(255, 255, 255, 0.5);
+  z-index: 1;
+}
+.tool-thumbnail:hover::before {
```
Comment (mabdurrahman) [inline assets/css/styles.css:55]: I can't see this in action, are you sure it's working as intended?

File: package.json [hunk 3]
```diff
@@ -0,0 +1,24 @@
+{
+  "name": "mabdurrahman.github.io",
+  "version": "1.0.0",
+  "description": "> This is my Resume.",
+  "main": "index.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
```
Comment (mabdurrahman) [inline package.json:7]: Can we have a script "start" to start Grunt? or should I only use the Grunt command?

File: Gruntfile.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
+'use strict';
```
Comment (mabdurrahman) [inline Gruntfile.js:1]: i tried "grunt connect" from Terminal the server started but don't handle the index.html page, am I using it in the correct way?

File: .gitignore (hunk 1 above)
Comment (AmrAbdulrahman) [reply .gitignore:1]: But this is project based configurations. The `node_modules/` directory is belonging to the project itself.
The global root .gitignore should have auto-generated system files, `*.DS_Store` for example or compiled files like (`.exe`, or `/build` and `/dist` directories)
You may need to keep it tracked in another project.

**Review by mabdurrahman — COMMENTED** [review]
@AmrAbdulrahman i'm sorry I didn't realize that I don't have .gitignore already on project root, and thought that you put another .gitignore to a subdirectory...

Just check the below comments and let me know if you gonna address something or not?

**Pull request merged**